Refactoring

ChrisMuller80 opened the pull request:
**Fixes** 

[DD-169](https://drivenbydata.atlassian.net/browse/DD-169) - Develop BRI-side code/service for push-based registration of NBNs to BRI

**How to test**

- check out this PR
- in local MYSQL instance, create database using the sql script located at `database/nbnresolver_updated.sql`
- in MYSQL execute following commands:
`INSERT INTO registrant (registrant_groupid) VALUES ('dans');`
`INSERT INTO credentials (registrant_id, org_prefix, username, password) VALUES (LAST_INSERT_ID(), 'UI:13', 'dansusername', 'cnSnMTIo' );`
- run `mvn clean package jetty:run`
- copy applications.properties file from `src/test/resources/application.properties` to `src/main/resources`
- fill in MYSQL credentials in `src/main/resources/application.properties`
- go to `http://localhost:8009/swagger-ui`
- get api token with credentials from db by using the `POST /token` endpoint
- enter the returned token in the `BearerAuth` field under the `Authorize` button at the top of the page
- start with a `POST /nbn` request

wilkos-dans commented on openapi.yaml:
```diff
@@ -0,0 +1,262 @@
+openapi: 3.0.3
```
Deze file verwacht ik niet in de root?!

wilkos-dans reviewed (commented):
* Kunnen onze eigen implementatie classes niet in een package met een andere naam? Ik mis iets van: nl.knaw.dans.nbnresover?
 * Ik mis de swagger-ui files onder de webapp.

ChrisMuller80 commented:
>     * Kunnen onze eigen implementatie classes niet in een package met een andere naam? Ik mis iets van: nl.knaw.dans.nbnresover?
> 
>     * Ik mis de swagger-ui files onder de webapp.

Done

wilkos-dans reviewed (changes requested):
Hi,
Please notice:
- I updated the POM, to include DANS parent POM now.
- I am missing the src/main/resources folder? (`Caused by: java.util.MissingResourceException: Can't find bundle for base name application, locale en_N`L)
- I am missing a unit test.

ChrisMuller80 commented:
> Hi,
> Please notice:
> 
>     * I updated the POM, to include DANS parent POM now.
> 
>     * I am missing the src/main/resources folder? (`Caused by: java.util.MissingResourceException: Can't find bundle for base name application, locale en_N`L)
> 
I changed this. The resource bundle was not needed.
>     * I am missing a unit test.

wilkos-dans commented:
Still in Draft/WIP?

wilkos-dans commented:
Looks good.
One more comment for now: If the DB (connection) is missing/DB stops, (almost?) all the API calls return this json:
```
{
  "code": 401,
  "message": "Authentication information is missing or invalid"
}
```
Which is not correct, or at least does not reveal the 'real' problem.
